src/components/AppModal.vue

```diff
@@ -0,0 +1,152 @@
+<template>
+  <div class="container" :class="{ inactive: !isModalActive }">
+    <div class="header">
+      <h2>{{ modalTitle }}</h2>
+      <i class="fa-solid fa-x" @click="toggleAppModal"></i>
```

Comment:
Wrap this icon with a `button` tag, every clickable item that has some action should be a button ,
and add the click event to the button